### PR TITLE
Add qsscheck to CI pipeline and improve .travis.yml config

### PR DESCRIPTION
**Pull request by Holzhaus**
This adds a simple checker script for `*.qss` files to the `scripts/` folder and runs it on Travis CI builds. It also cleans up the `.travis.yml` file a bit.

### CODE REVIEW

**Comment by uklotzde** [comment]
Please check for trailing whitespaces that CodeFactor complains about.

**Comment by Holzhaus** [comment]
> Please check for trailing whitespaces that CodeFactor complains about.

Ok, done.

**Comment by Holzhaus** [comment]
By the way, the `qsscheck` job currently fails, but the issues are fixed by PRs #2374 and #2342.

**Comment by ywwg** [comment]
checking by travis is good, but is there a way we can integrate it into mixxx-test?  even just having a simple C++ test file that shells out to python could work.

File: scripts/qsscheck.py
Comment (ywwg) [inline scripts/qsscheck.py]: it would be nice to reduce some of the indentation in this file.  In this case, you can do:

```
if os.path.splitext(fname)[1] != '.xml':
  continue
```

And save one level

File: scripts/qsscheck.py
Comment (ywwg) [inline scripts/qsscheck.py]: Can you factor out all the regexes into top-level global constants and then refer to them by name where they are used? I see that they are only used once, but that pattern makes it easier to understand what each regex is doing and why it's being used

File: scripts/qsscheck.py
Comment (ywwg) [inline scripts/qsscheck.py]: can you add docstrings (https://www.python.org/dev/peps/pep-0257/) to these functions?  They have pretty complex yielding behavior that warrants explanation.

File: scripts/qsscheck.py
Comment (ywwg) [inline scripts/qsscheck.py]: (see below comment -- here's another place where we can save a level of indentation)

**Review by ywwg — CHANGES_REQUESTED** [review]
Just some python style nits. Thanks for this useful utility!

File: scripts/qsscheck.py
Comment (Holzhaus) [reply scripts/qsscheck.py]: Ok

**Comment by Holzhaus** [comment]
> @ywwg: checking by travis is good, but is there a way we can integrate it into mixxx-test? even just having a simple C++ test file that shells out to python could work.

What's the benefit? For PRs that only make skin changes, you'd have to wait until Mixxx is built (2 minutes vs >30min). Also, this would also introduce a dependency on Python/PyQt/tidycss, which means that we need to install them on OSX and Appveyor, too. That would increase the likelyness of running into a timeout.

**Comment by ywwg** [comment]
>  introduce a dependency on Python/PyQt/tidycss

If people want to run the script they will need these things anyway, right?

But I think it's ok to submit without having test integration. But we should make sure to update the skinning documentation to call out this script so people know about it.

File: scripts/qsscheck.py
Comment (Holzhaus) [reply scripts/qsscheck.py]: How? We have an elseif case, too.

**Comment by Holzhaus** [comment]
I'll update the wiki after merge (since the tool is not in the scripts directory without this PR).

Is there something missing or can this be merged as soon as #2374 and #2342 have landed in master?

**Comment by Holzhaus** [comment]
Let's wait for the CI builds. If we have a clean state and nobody raises any concerns, I'll merge this tomorrow, so that this can prevent introducing new issues in the other pending skin PRs.

**Pull request merged**
